src/plugins/share/public/components/url_panel_content.tsx

```diff
@@ -428,6 +388,37 @@ export class UrlPanelContent extends Component<UrlPanelContentProps, State> {
     }
   };
 
+  private renderCopyButton = () => (
+    <EuiCopy
+      beforeMessage={this.state.showWarningButton ? this.props.snapshotShareWarning : undefined}
+      textToCopy={this.state.url || ''}
+      anchorClassName="eui-displayBlock"
+    >
+      {(copy) => (
+        <EuiButton
+          fill
+          fullWidth
+          onClick={copy}
+          disabled={this.state.isCreatingShortUrl || this.state.url === ''}
+          data-share-url={this.state.url}
+          data-test-subj="copyShareUrlButton"
+          size="s"
+          iconType={this.state.showWarningButton ? 'alert' : undefined}
```

Comment:
Basing iconType and color on just `this.state.showWarningButton` and not `this.state.showWarningButton && this.props.snapshotShareWarning` results in button always showing the warning even when there are no unsaved changes to the dashboard. The screen shot below, notice that the dashboard is in view mode but the button is displayed in warning state. I would recommend not adding `showWarningButton` to state and just moving into the function `renderCopyButton` and then using `showWarningButton` to toggle message, icon, and color.
```
const showWarningButton =
      this.props.snapshotShareWarning &&
      this.state.exportUrlAs === ExportUrlAsType.EXPORT_URL_AS_SNAPSHOT;
```

<img width="300" alt="Screen Shot 2023-02-16 at 2 42 44 PM" src="https://user-images.githubusercontent.com/373691/219493788-08646a92-61f6-4cc4-88e5-f90c5131d40b.png">

Reply:
Ahh, you're totally right! Not sure how I managed to drop that part of the conditional 🤦 Fixed in 42c073ebfb3da82d4f656afb6192f8c8e2b99123 - thanks for catching that!